### ISSUE TITLE
'undefined' rtsp stream URL

### ISSUE THREAD

**Issue by tmur11** [hjdhjd/homebridge-unifi-protect2#49]
I have an issue with RTSP stream URL. Hostname/IP appears as undefined. Video stream will not work but still image does.

![image](https://user-images.githubusercontent.com/21132832/91249791-8d616d00-e725-11ea-83a7-27efcddbdc37.png)

![image](https://user-images.githubusercontent.com/21132832/91249857-a833e180-e725-11ea-8b2f-d4cfa0daf8bd.png)

**Comment by hjdhjd** [comment]
The plugin doesn't set the RTSP stream, simply enable it. The above is a UniFi Protect issue, not one related to `homebridge-unifi-protect2`. The error it's generating is actually correct...it's trying to resolve the RTSP stream it's pulling off of Protect.

Pulling image snapshots has nothing to do with the RTSP stream, which may seem counterintuitive. :smile: The snapshots are pulled from the Protect controller, not the camera.

TL;DR - fix the RTSP stream on correct and the plugin will pull it. I would try, in this order:

1. Disabling and reenabling RTSP to see if it gets it right.
2. Unmanagaing and remanaging the camera, and see if that works.
3. Filing a support request with Ubiquiti.

Good luck.

**Comment by tmur11** [comment]
thanks for the quick response. I realize it seems to be an issue with the protect interface but the actual stream works fine.

I have tried your suggestions and aside from building a new protect controller VM I was hoping we could somehow build the stream URL from the hostname/IP provided in the config.json file.

I have raised issue with unifi but I am running the a older version of protect (12.5) on unsupported hardware so I doubt I'll get a response.

**Comment by hjdhjd** [comment]
As far as I know, you can't run Protect controllers in a VM. It needs to be on a UniFi-sanctioned device...but point me in the right direction if that's not the case.

I don't intend to support overriding pulling the configuration from config.json right now. Might put it on the backlog...but if you want to specify your own parameters and configuration, you're probably better off with homebridge-camera-ffmpeg. The North Star here is to seamlessly complement Protect. If your Protect setup is faulty...that's your root cause here I'm afraid.

Good luck.

**Comment by tmur11** [comment]
Running protect in a VM isn't supported but a DEB package was released/leaked by unifi (v12.5) which can be still found.

https://www.reddit.com/r/Ubiquiti/comments/dchpv5/diy_ubiquiti_protect_for_linux/?utm_source=share&utm_medium=web2x&context=3

thanks for the help would just be nice not having to enter all my cameras manually into homebridge-camera-ffmpeg. I will try a fresh controller install and see what results I get.

thanks again

**Comment by hjdhjd** [comment]
Fascinating. Yeah...you're way off the reservation and definitely on your own in uncharted waters there. 

Glad you enjoy the plugin...my suggestion, for what it's worth...buy a controller. They aren't super-pricey (UCK Gen2+ is ~$200) relative to the cost of some of the cameras they've got. :smile: It's the right way to do it, and medium/long-term it will save you hours of heartache trying of going back and forth trying to maintain an unsupported (and at this point quite dated) Protect controller release. Just my $.02 here...
